### DISCUSSION TITLE
Looking to create tools (Similar to custom editors or custom inspectors)

### DISCUSSION THREAD

**Discussion by CWilliamsDev** [stride3d/stride#1970] [Newcomers] [answered]
Hello! Firstly, I wanted to thank everyone who has contributed to Stride over the years before I discovered it.

In Unity, I tend to use a lot of custom editor windows. For most projects I created at least 2 or 3 to help designers quickly edit or create objects within the game's data model. (Quest Editor, Path Scripting for NPCs, etc).

I'd like to be able to continuing doing this in Stride, so my questions are:

- Is there out of the box functionality for these custom editor tools, or adding new windows in the editor, or will I have to do this from source? If it does need to be done from source, is there a good resource that can get me started with best practises or syntax?

- I don't believe there is anything similar to Scriptable Objects, will I likely be dumping data from these tools into json files?

**Comment by Doprez** [comment, marked as answer]
unfortunately extending custom editor windows is currently not a thing in Stride. The closest thing is using EntityProcessors to create custom tools within the scene view but it has its limitations.
I did make a [brief overview](https://www.youtube.com/watch?v=fk8aH2AY2TI&t=1s) of using EntityProcessors for this if you would like to take a look.

Currently if you wanted to extend the editor you would have to do it by running the source code locally and building off of the existing WPF UI. I believe there is an effort from some in the Discord to create this need but I dont believe its close to done.

**Reply by Doprez** [reply]
Also for your second question there are a couple ways to store data.
- Json/Yaml like you said
- using GameSettings by [inheriting from Congiuration](https://www.youtube.com/watch?v=QKizq1xC4vk)
- using stored assets or prefabs by loading from the [content db](https://doc.stride3d.net/latest/en/tutorials/csharpbeginner/loading-content.html)

**Reply by CWilliamsDev** [reply]
Thank you. This is much appreciated. I'll take a look at EntityProcessors before diving into source.
